src/modules/roc_pipeline/converter_sink.h

```diff
@@ -57,13 +57,13 @@ class ConverterSink : public sndio::ISink, public core::NonCopyable<> {
 private:
     audio::NullWriter null_writer_;
 
-    core::ScopedPtr<audio::PoisonWriter> resampler_poisoner_;
-    core::ScopedPtr<audio::ResamplerWriter> resampler_writer_;
-    core::ScopedPtr<audio::IResampler> resampler_;
+    core::Optional<audio::PoisonWriter> resampler_poisoner_;
+    core::Optional<audio::ResamplerWriter> resampler_writer_;
+    core::Optional<audio::IResampler> resampler_;
```

Comment:
@gavv Just another question, thanks for you review. 

In this line `core::Optional<audio::IResampler> resampler_;` i should not use **core::Optional** as **audio::IResampler**  interface right ?

Reply:
Yep.